slicer-plugin/NvidiaAIAA/SegmentEditorNvidiaAIAALib/SegmentEditorEffect.py

```diff
@@ -389,21 +390,24 @@ def onClickSegmentation(self):
             if self.updateSegmentationMask(extreme_points, result_file, modelInfo):
                 result = 'SUCCESS'
                 self.updateGUIFromMRML()
-        except AIAAException:
-            self.closeAiaaSession()
-            slicer.util.warningDisplay(operationDescription + " - session expired.  Retry Again!")
-            return
+                self.onClickEditAnnotationFiducialPoints()
+        except AIAAException as ae:
+            logging.exception("AIAA Exception")
+            if ae.error == AIAAError.SESSION_EXPIRED:
+                self.closeAiaaSession()
+                slicer.util.warningDisplay(operationDescription + " - session expired.  Retry Again!")
+            else:
+                slicer.util.errorDisplay(operationDescription + " - " + ae.msg, detailedText=traceback.format_exc())
         except:
+            logging.exception("Unknown Exception")
             slicer.util.errorDisplay(operationDescription + " - unexpected error.", detailedText=traceback.format_exc())
-            return
         finally:
             qt.QApplication.restoreOverrideCursor()
             self.progressBar.hide()
 
         msg = 'Run segmentation for ({0}): {1}\t\n' \
               'Time Consumed: {2:3.1f} (sec)'.format(model, result, (time.time() - start))
         logging.info(msg)
-        self.onClickEditAnnotationFiducialPoints()
```

Comment:
why do we remove this,
or we put it somewhere else?

Reply:
fixed

Reply:
it's moved after success